addons/phobos/src/main/scala/com/ocadotechnology/pass4s/phobos/syntax.scala

```diff
@@ -100,4 +102,14 @@ object syntax {
       consumer.mapM(XmlDecoder[A].decode(_).liftTo[F])
   }
 
+  implicit final class ConsumeXmlGenericMessageSyntax[F[_], A](private val consumer: Consumer[F, A]) {
+
+    def asXmlConsumer[B: XmlDecoder](
+      implicit M: MonadError[F, _ >: ru.tinkoff.phobos.decoding.DecodingError],
```

Comment:
I'm assuming you didn't change `ConsumeXmlMessageSyntax` to also require such `MonadError` instance to avoid breaking binary compatibility?

Reply:
That's true, I thought it's worth to keep it binary compatible